component/newstorage/formattedstore/formattedstore.go

```diff
@@ -175,6 +195,23 @@ func (f *FormattedProvider) GetStoreConfig(name string) (newstorage.StoreConfigu
 	return newstorage.StoreConfiguration{TagNames: tagNames}, nil
 }
 
+// GetOpenStores returns all currently open stores.
+func (f *FormattedProvider) GetOpenStores() []newstorage.Store {
+	openStores := make([]newstorage.Store, len(f.openStores))
+
+	var counter int
+
+	f.lock.Lock()
```

Comment:
Lock should be at the top since you're referencing f.openStores. Also, this should be a read lock RLock().

Reply:
Done.

Reply:
Also fixed the same mistake I made in batchedstore.go